lib/chess_club/user_manager/context.ex

```diff
@@ -0,0 +1,27 @@
+defmodule ChessClub.UserManager.Context do
+  @behaviour Plug
+
+  import Plug.Conn
+
+  def init(opts), do: opts
+
+  def call(conn, _) do
+    context = build_context(conn)
+    Absinthe.Plug.put_options(conn, context: context)
+  end
+
+  def build_context(conn) do
+    with ["Bearer " <> token] <- get_req_header(conn, "authorization"),
+         {:ok, current_user} <- authorize(token) do
+      %{current_user: current_user}
+    else
+      _ ->
+        %{}
+    end
+  end
+
+  defp authorize(token) do
+    token
+    |> ChessClub.UserManager.Guardian.decode_and_verify(%{"typ" => "access"})
```

Comment:
Generally only pipe when there are 2+ pipes.

Reply:
There is a check somewhere we can implement if we want that. Credo is configurable.